src/analyses/call_graph_helpers.cpp

```diff
@@ -102,6 +102,15 @@ std::set<irep_idt> get_functions_reachable_within_n_steps(
     graph, start_functions, n, forwards);
 }
 
+void disconnect_unreachable_functions(
+  call_grapht::directed_grapht &graph,
+  const irep_idt &function)
+{
+  std::size_t start_idx = *(graph.get_node_index(function));
+  if(start_idx != optionalt<std::size_t>())
```

Comment:
If it was Empty like this the deref on the line above would already have faulted. This should probably be
```
if(auto start_idx = graph.get_node_index(function))
  graph.disconnect_unreachable(*start_idx);
```

(This line compiled because `optionalt<std::size_t>` has an implicit conversion from `std::size_t`, so the current test is equivalent to `if(optionalt<std::size_t>(start_idx) != optionalt<std::size_t>())`, which is always true)

Reply:
The `*` operator would cause a failing assertion. If you want to do a check, don't apply `operator*` and test `.has_value()` of the result before using (i.e., dereferencing).

Reply:
So, if I want to throw an error anyway, I may as well just do what is done in the get_reachable function?